pkg/kubelet/nodeshutdown/nodeshutdown_manager_linux.go

```diff
@@ -246,60 +262,125 @@ func (m *Manager) processShutdownEvent() error {
 	klog.V(1).InfoS("Shutdown manager processing shutdown event")
 	activePods := m.getPods()
 
-	nonCriticalPodGracePeriod := m.shutdownGracePeriodRequested - m.shutdownGracePeriodCriticalPods
+	groups := groupByPriority(m.podPriorityShutdownGracePeriod, activePods)
```

Comment:
These changes are not feature flagged... how do we want this to behave if the new feature is disabled?